Bug 6625 - Dodanie testów

DBrzoskowski opened the pull request:
https://bugzilla.stxnext.pl/show_bug.cgi?id=6625

Commit:
testy_poczatek

Commit:
testy

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -7,7 +7,9 @@
-from presence_analyzer import main, views, utils
+import main
+import views
+import utils
```
Sort imports.

dpolcyn commented on src/presence_analyzer/views.py:
```diff
@@ -6,8 +6,8 @@
-from presence_analyzer.main import app
-from presence_analyzer.utils import jsonify, get_data, mean, group_by_weekday
+from main import app
+from utils import jsonify, get_data, mean, group_by_weekday
```
Sort imports.

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -86,6 +109,51 @@ def test_get_data(self):
+        start_date = datetime.time(10, 10, 10)
+        end_date = datetime.time(12, 10, 10)
+        date = utils.interval(start_date, end_date)
+        self.assertLess(start_date, end_date)
```
You don't check return.

Commit:
testy_poprawki

testy_porawione_uwagi

testy_poprawki

testy_porawione_uwagi

testy_poprawki

testy_porawione_uwagi

testy_poprawki

testy_porawione_uwagi

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -86,6 +110,60 @@ def test_get_data(self):
+        Test group by weekend
+        """
+        sample_date = datetime.date(2013, 9, 12)
+        d = {sample_date: {'end': datetime.time(23, 23, 51),
```
Maybe `day` instead of d?

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -86,6 +113,60 @@ def test_get_data(self):
+        """
+        Test calculates arithmetic mean.
+        """
+        lista = [24261, 32533, 34323, 32453, 54154, 32412, 31123]
```
`Lista` is a polish word.

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -86,6 +113,60 @@ def test_get_data(self):
+        start_date = datetime.time(8, 31, 6)
+        end_date = datetime.time(15, 15, 27)
+        date = utils.interval(start_date, end_date)
+        self.assertLess(start_date, end_date)
```
You don't check anything expected here.

dpolcyn commented on src/presence_analyzer/views.py:
```diff
@@ -50,7 +50,7 @@ def mean_time_weekday_view(user_id):
         for weekday, intervals in enumerate(weekdays)
     ]
-
+    # import ipdb; ipdb.set_trace()
```
????

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -51,10 +55,58 @@ def test_api_users(self):
+        ]
+
+        self.assertEqual(data, expected_list)
+        self.assertEqual(len(data), 7)
```
You don't have to check it. You already did it line up.

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -51,10 +55,58 @@ def test_api_users(self):
+
+        self.assertEqual(data, expected_list)
+        self.assertEqual(len(data), 7)
+        self.assertIsInstance(data, list)
```
You don't have to check it. You already did it 2 lines up.

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -84,8 +132,101 @@ def test_get_data(self):
+        self.assertEqual(len(data), 7)
+        self.assertEqual(data, excepted_result)
+        self.assertTrue(datetime.datetime.strptime(
+            str(datetime.date(2013, 9, 12)), '%Y-%m-%d'))
```
Style

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -84,8 +132,101 @@ def test_get_data(self):
 
+    def test_group_by_weekday(self):
+        """
+        Test group by weekend
```
Missing full stop.

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -84,8 +132,99 @@ def test_get_data(self):
+    def test_interval(self):
+        """
+        Test interval between two datetime.time objects.
+
```
You don't need this empty line.

dpolcyn commented on src/presence_analyzer/tests.py:
```diff
@@ -86,6 +132,95 @@ def test_get_data(self):
+        Test group by weekend.
+        """
+        excepted_result = [[], [0], [17088], [10800], [], [], []]
+        days = {
```
You don't have to declare this. Put it in utils.group_by_weekday because you use it only once.
